File: fsharp/armstrong-numbers/ArmstrongNumbers.fs
module ArmstrongNumbers

open System

let isArmstrongNumber (number: int): bool = 
    let (digits: int list) = number |> string |> Seq.toList |> List.map Char.GetNumericValue |> List.map int
    let pow = digits.Length

    List.sum [ for d in digits -> pown d pow ] |> (=) number

File: java/armstrong-numbers/src/main/java/ArmstrongNumbers.java
import java.util.stream.*;

class ArmstrongNumbers {

	IntStream getDigits(String n) {
		return n.chars()
			.map(c -> Character.getNumericValue(c));
	}

	boolean isArmstrongNumber(int n) {
		String str = Integer.toString(n);
		double pow = str.length();
		return n == str
			.chars()
			.map(c -> Character.getNumericValue(c))
			.map(d -> (int)Math.pow((double)d, pow))
			.sum();
	}

}
